### ISSUE TITLE
Upgrade neighbourhood shapes, names for Cleveland

### ISSUE THREAD

**Issue by skorasaurus** [whosonfirst-data/whosonfirst-data#189]
### tl;dr

Status:
- [x] Propose **sources**
- [x] Establish **concordances** between WOF and source
- [x] **Upgrade / add** WOF shapes
- [x] **Add new or missing** features to WOF
- [x] **Deprecate / supersede** WOF features
- [x] **Draft review**
- [x] **Draft revisions** based on review comments
- [ ] **Final review**
- [ ] **Final revisions** based on review comments
- [ ] **Import** into Who's On First

### Details

Sourcing:
- Permissive license?
- Add new src to Who's On First

Upgrade / add:
- `macrohood`
- `neighbourhood`
- `microhood`

Add new or missing features to WOF
- `macrohood`
- `neighbourhood`
- `microhood`

Deprecate / supersede:
- `macrohood`
- `neighbourhood`
- `microhood`

Original report:
---

Hi, 

[Several neighbourhoods in Cleveland, Ohio in the wof-data](https://whosonfirst.mapzen.com/spelunker/id/101712563/descendants/?placetype=neighbourhood&page=1) aren't classified correctly and/or their geometry is incorrect. 

A couple examples (an incomplete list): 

[South Brooklyn's geometry](https://whosonfirst.mapzen.com/spelunker/id/85849511/) is actually a part of [Old Brooklyn](https://whosonfirst.mapzen.com/spelunker/id/85871819/) 
[Chinatown](https://whosonfirst.mapzen.com/spelunker/id/85871855/) is known as Asiatown. 
[The West Bank's geometry](https://whosonfirst.mapzen.com/spelunker/id/420534715/) is incorrect. 

The West Bank, The East Bank, Playhouse Square, The Warehouse District, [Gateway District](https://whosonfirst.mapzen.com/spelunker/id/85871849/), and the [Civic Center](https://whosonfirst.mapzen.com/spelunker/id/85871827/) possibly could be classified as microhoods, under [The Downtown Neighborhood](https://whosonfirst.mapzen.com/spelunker/id/85867113/)

I'd be happy to give a much more thorough and complete breakdown and submit a pull request but I haven't figured out how to do it without downloading the very large files. Thanks to @thisisaaronland who is working on this by making smaller extracts (possibly related to [#188](https://github.com/whosonfirst/whosonfirst-data/issues/188).

**Comment by thisisaaronland** [comment]
Hi Will,

The work on per-placetype tarball/bundles is chugging along nicely so there
should be something to poke at shortly.

In the meantime the raw GeoJSON for each record is linked to from its
permalink on the spelunker.

For example, this:

https://whosonfirst.mapzen.com/spelunker/id/85682513/

Links to this:

https://whosonfirst.mapzen.com/data/856/825/13/85682513.geojson

If you know the ID of a record you can always grab the source from the
domain above. The details of converting an ID to a URL are outlined on:

https://whosonfirst.mapzen.com/data

**Comment by thisisaaronland** [comment]
@skorasaurus In the hiding-in-plain-sight and if-youd-like-to-help-us-kick-the-tires department we now have per-placetype "bundles" of WOF data available for downloading as handy tarballs:

https://whosonfirst.mapzen.com/bundles/

This doesn't solve the larger issue of SO MUCH GIT to wrangle so, in the short-term, edits will need to be appended and we can take it from there. It's not perfect but this true of many things...

**Comment by nvkelso** [comment]
**Other possible sources:**

- Wikipedia: 
  - https://en.wikipedia.org/wiki/Cleveland#Neighborhoods 
- Good narrative: 
  - http://www.streetdirectory.com/travel_guide/209273/north_america_destinations/cleveland_ohio_neighborhoods.html 
- official Statistical Planning Areas (they are in fact used)
 - http://planning.city.cleveland.oh.us/gis/cpc/basemap.jsp?command=larger
- In PDF form: http://planning.city.cleveland.oh.us/maps/jpg/All_SPAs_2014.jpg 
- See long list of local maps of various kinds at: 
  - http://planning.city.cleveland.oh.us/maps/maplist.php?pageNum_rs1=0&totalRows_rs1=155 
- http://www.rustbelt.com/portfolio/Downtown_2007.pdf

**Comment by nvkelso** [comment]
From @skorasaurus comment https://github.com/whosonfirst-data/whosonfirst-data/issues/404#issuecomment-236710351:
---

ref: 
https://github.com/whosonfirst-data/whosonfirst-data/issues/189

I've been in touch with @stepps00 on it in the past couple weeks and gave me the starter kit for it. 

(Some of my below includes my response to him in an email): 

The city of Cleveland does not have any defined neighbourhood boundaries (only statistical planning areas that are mostly used by the City Planning department). 

I actually created some of my own neighborhood boundaries a couple years ago (I hadn't heard of alpha or betashapes) - https://skorasaurus.wordpress.com/cleveland-neighborhood-map/ that I synthesized from a variety of sources, personal knowledge and asking residents and CDCs. 

Since that last published release of the work, I've lived here longer and would feel much more comfortable expanding that data although I wouldn't feel extremely confident on defining boundaries for some parts of the city. In places with those doubts, I can rely upon the existing WOF shapes.  

If you're comfortable and A-OK with that, I can go ahead. 

Lastly, what's your timeframe on it to have it completed?

**Comment by nvkelso** [comment]
Hi @skorasaurus, I was cleaning up a backlog of issues and forgot you had already opened one on Cleveland, sorry about spamming you!

Yes, we'd love your help with fixing up your city! We've refined the process a bit so please get another export from @stepps00 to minimize merge conflicts.

Timeframe is flexible, but before NACIS? Say by end of August?

**Comment by stepps00** [comment]
Hey @skorasaurus! As Nathaniel mentioned, we've refined our neighbourhood update process a bit - I've attached a more recent .geojson file in this .zip file: [Cleveland.zip](https://github.com/whosonfirst-data/whosonfirst-data/files/397956/Cleveland.zip)

The .geojson is similar to the file I previously sent you, though this .geojson file only includes attribute fields relevant to neighbourhood upgrades (it does not include variant or preferred names and other irrelevant information). I'll follow up with you in an email; let me know if you have any questions!

**Comment by skorasaurus** [comment]
I merged my source file and the wof file that Stephen emailed me on 2016-08-02 into a single geojson file and will track my edits at https://github.com/skorasaurus/quiencle 
because the whosonfirst-data repo is quite large (of data) and complicated to locally fork. 

Questions: 
Do I need to create my own brooklyn integers for all of the places in [my source](https://github.com/skorasaurus/cleboundaries/tree/master/neighborhoods)? I am aware that I'll need to create the brooklyn integers for places that will be new to WOF that I add to WOF. Each record in my source do have a unique ID but I don't recall the reason that I had created the IDs, they may be an artifact from originally editing the files in geojson.io or josm.

**Comment by nvkelso** [comment]
Yes, you need Brooklyn Integers for all your new places and existing Brooklyn Ints on modified places.

So for a feature that has an id of "way-1234" in both the feature id and the property hash id, you'll need (the same) Brooklyn Int there instead.

For modifications of WIOF records, just copy over the existing id to your new feature.

**Comment by skorasaurus** [comment]
Can I assume the information on the wiki is still best practice, regarding deprecating records? 
https://github.com/whosonfirst/whosonfirst-docs/wiki/What-is-a-Significant-Event%3F

**Comment by nvkelso** [comment]
Yes! 

(There is one minor point Stephen will clarify next week.)

**Comment by stepps00** [comment]
Hey @skorasaurus!

The information regarding deprecating records on our wiki page is still valid, with one minor edit (as Nathaniel mentioned).

Instead of updating the `edtf:deprecated` field, we want to update the `edtf:cessation` field. 
In addition, we want to add the following (field and value) to the record:

`"mz:is_current":0,`

I'll be updating the wiki and tutorial today with current best practices - let me know if you have any questions.

**Comment by skorasaurus** [comment]
@stepps00 ; the geojson you gave me (2016-08-02) does not contain a cessation field. Should I create a new field (named cessation) for it? 

My current edits is at https://github.com/skorasaurus/quiencle/commits/maste rbut it is admittedly a little sloppy: After editing a couple neighbourhoods' geometries, I learned that I needed to create new geometries for them because they contained significant events (instead of editing them).

**Comment by stepps00** [comment]
Good question - are you planning on submitting the .geojson file with edits independently of a PR?

If so, since I already gave you a .geojson file that didn't contain the `edtf:cessation` field (or `mz:is_current` field), go ahead and continue with your edits and add a cessation field. Once you submit your .geojson file, I'll go ahead and update the `cessation` and `is_current` fields accordingly.

**Comment by skorasaurus** [comment]
@stepps00 

Likely submitting the geojson file that includes the edits independent of a PR instead of a PR because the repository is relatively large (in data size), and I hadn't been able to fork it.

**Comment by stepps00** [comment]
OK - since you're neighbourhood updates are in progress, go ahead and use the `edtf:deprecated` field. I can update the field name on import. :)

**Comment by skorasaurus** [comment]
Sounds good; still coming along. Had a question regarding names: 

What's the guidelines for abbreviations? [This](https://whosonfirst.mapzen.com/spelunker/id/85835641/#14/41.4643/-81.5899) is the record in question and wanted to confirm whether this practice (using eng_x_preferred and name to the abbreviated version; the variant to the spelled out version) is correct.

- 

-

**Comment by stepps00** [comment]
Hey @skorasaurus, good question! 

We don't have guidelines on what would constitute a "correct" abbreviation; Who's on First has both the abbreviation [Mt.](https://whosonfirst.mapzen.com/spelunker/id/85835487/#13/37.6349/-122.1290) and [Mount](https://whosonfirst.mapzen.com/spelunker/id/85835737/#14/36.1729/-85.5208) spelled out for records' `wof:name`.  As long as "Mount Pleasant" is preserved as an `eng_x_preferred` name, "Mt Pleasant" should be fine as the `wof:name` (which would appear on our map labels and in search results).

**Comment by skorasaurus** [comment]
Hi, here's a first draft! - https://github.com/skorasaurus/quiencle
I got bite by a GDAL bug that delayed me for a couple days along with other things. 

These include my edits based upon a WOF geojson file from [wof-2016-08-02.geojson.zip](https://github.com/whosonfirst-data/whosonfirst-data/files/443480/wof-2016-08-02.geojson.zip) @stepps00 gave to me. I've attached it to here as reference, zipped, in the above link, for comparison. 

A couple questions about my work so far: 
- min_zoom? Are you setting this subjectively? I started to set some of them subjectively myself, then decided to leave them NULL half way and wait for your feedback. 

- I wasn't sure what array syntax to use as mentioned in the [superceded_by and superceded](https://github.com/whosonfirst/whosonfirst-properties/blob/56bcf9f4f4905b4fb5d4724a7cd16e270a7a19d6/properties/wof.md) notes. With records that have multiple supercede/superceded_by values, I left them blank and await your suggestion. (Perhaps the format is [123, 762] as used in the property of [u'wof:belongsto':](https://whosonfirst.mapzen.com/spelunker/id/101712563/#10/41.4976/-81.7059) in Cleveland's raw file)? 

Other notes: 

- The deprecated column: records with a value of '1' have been superceded by other records. -1 means no significant events happened (only minimal geometry editing) to the record or it is a new record. 
I used this column as I went along in my edits whether or not to know that I've edited it. (If I were to do this again, I'd use a different column, perhaps is_funky, for clarity's sake as there were several records which I edited slightly but didn't have designate it as 1). 

- I wasn't sure what to do ['is_funky'](https://github.com/whosonfirst/whosonfirst-properties/blob/56bcf9f4f4905b4fb5d4724a7cd16e270a7a19d6/properties/mz.md) column. I left it null in my created records. 

- Also not sure on the [is_landuse_aoi](https://github.com/whosonfirst/whosonfirst-properties/blob/56bcf9f4f4905b4fb5d4724a7cd16e270a7a19d6/properties/mz.md). I left it null or blank. 

- is_hard (was included in my original as is_hard_boundary), eng_variant_name, and is_official. Some of the values were 0 or -1; I usually copied the value from the superceded record or left it as null. 

- tier_locality isn't mention in [WOF properties](https://github.com/whosonfirst/whosonfirst-properties), I usually copied the value from the superceded record or left it as null. 

- A couple of my columns have values for eng_preferred_name. This occurred when I manually copied and pasted the raw geojson from the wof-2016-08-02.geojson into my whosonfirst.geojson file. I had to do that a couple times because I had inadvertently modified the WOF geometry and then committed it onto github. I don't know why, but these values weren't being recorded when I edited them in qgis. 

There were a couple records in the WOF records that were actually apartment buildings and classified them as [venue](https://github.com/skorasaurus/quiencle/issues/1). In both instances, their locality_id record also needed to be changed.

**Comment by stepps00** [comment]
Hey @skorasaurus - thanks for the work so far! I've tried my best to answer your questions and give comments below (your initial comments are in _italics_):

 _min_zoom? Are you setting this subjectively? I started to set some of them subjectively myself, then decided to leave them NULL half way and wait for your feedback._

The [second part of our neighbourhood tutorial](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/issue_workflows/sf_neighbourhood_updates_pt_2.md) contains information on the `min_zoom` values. These are not subjective, as they impact map labelling; I've copied the information below:

* neighbourhood:
    * Default: min_zoom:15, optionally modify up to 13 for important places.
    * Default: max_zoom:18, rarely modify.

* macrohood:
    * Default: min_zoom:13, optionally modify up to 11 for important places.
    * Default: max_zoom:15, rarely modify.

* microhood:
    * Default: min_zoom:16, rarely modify.
    * Default: max_zoom:18, rarely modify.

_I wasn't sure what array syntax to use as mentioned in the [superceded_by and superceded](https://github.com/whosonfirst/whosonfirst-properties/blob/56bcf9f4f4905b4fb5d4724a7cd16e270a7a19d6/properties/wof.md) notes. With records that have multiple supercede/superceded_by values, I left them blank and await your suggestion. (Perhaps the format is [123, 762] as used in the property of [u'wof:belongsto':](https://whosonfirst.mapzen.com/spelunker/id/101712563/#10/41.4976/-81.7059) in Cleveland's raw file)?_

The `supersedes`/`superseded_by` values would be entered as a list, just like [our Cleveland example](https://whosonfirst.mapzen.com/spelunker/id/101712563/#10/41.4976/-81.7059). So, your example format of [123, 762] would be correct. 

I am curious, though, about the features that have multiple `supersedes`/`superseded_by` values. Typically this is a 1:1 relationship. What is causing these records to have more than one `supersedes`/`superseded_by` value? If it's due to geographic overlap, we can go ahead and use a 1:1 relationship with the feature that shares the largest amount of overlap.

- _I'm not sure how you'd like to designate [Nela Park](https://github.com/skorasaurus/quiencle/issues/6)_

Typically, I would say Nela Park should be represented in WoF as a `microhood`. After researching, though, it looks like Nela Park is actually outside of the city boundary of Cleveland and a part of East Cleveland, OH. We can still import this record, but since it falls under a different `locality` (East Cleveland), we'd have to prepare some additional records for East Cleveland, OH. 

_Other notes:_ 

-_The deprecated column: records with a value of '1' have been superceded by other records. -1 means no significant events happened (only minimal geometry editing) to the record or it is a new record. _
_I used this column as I went along in my edits whether or not to know that I've edited it. (If I were to do this again, I'd use a different column, perhaps is_funky, for clarity's sake as there were several records which I edited slightly but didn't have designate it as 1)._

Makes sense. Typically when we import, we separate our .geojson files by placetype _and_ by whether or not the records are current. For San Francisco, for example, we have six geojson files to import: 

- SF_Invalid_Neighbourhood_Points.geojson _(Points that are no longer valid)_
- SF_Invalid_Neighbourhood_Polygons.geojson _(Polygons that are no longer valid)_
- SF_Macrohoods.geojson _(New macrohood records that used to be neighbourhood records)_
- SF_Microhoods.geojson _(New microhood records that used to be neighbourhood records)_
- SF_New_Neighbourhoods.geojson _(New neighbourhood records)_
- SF_Updated_Neighbourhoods.geojson _(Existing neighbourhood shapes that received updated geometries)_

_I wasn't sure what to do ['is_funky'](https://github.com/whosonfirst/whosonfirst-properties/blob/56bcf9f4f4905b4fb5d4724a7cd16e270a7a19d6/properties/mz.md) column. I left it null in my created records._

`is_funky` is used for labeling and display purposes. This field is reserved for features that are technically valid/correct, but we don't want them displayed on the map. If you've _created_ these records, then they should not be tagged as `is_funky`.

_Also not sure on the [is_landuse_aoi](https://github.com/whosonfirst/whosonfirst-properties/blob/56bcf9f4f4905b4fb5d4724a7cd16e270a7a19d6/properties/mz.md). I left it null or blank._ 

Similar to `is_funky`, `is_landuse_aoi` is used for labeling and display purposes. This is reserved for features that have a special landuse designation, such as a park. If the features you've created are actual neighbourhoods, you're fine leaving this field blank.

_There's the flats. It consists of two parts, the west bank and the East bank. I'd classify the east bank as being downtown but not the west bank, so should I give it two parent_id's?_

Instead of giving this record dual parent values, we should give it a value of `-3`.  Described in [our S3 requirements file](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/how_to/requirements_for_s3.md), a `-3` represents dual parent records.

Lastly, aside from breaking this file into several files (based on placetype and whether or not the value is current), we should make sure all geometries are snapped together and have valid geometries. I've noticed that the geometries of some of the new features do not have aligning geometries, which is something that should be taken care of prior to import.

 The work so far is **great** - thanks for working on this and let me know if you have any other questions or comments! 

cc @nvkelso.

**Comment by skorasaurus** [comment]
Thanks for the answers. I'll snap together the geometries in the next day or 2. 
Regarding the multiple superceded_by, it was because of geography in the instances that I remember. 

A couple more questions: 
min and max zooms for the venues? (There were 2 neighbourhoods in the WOF that I've reclassified as venues, RE: https://github.com/skorasaurus/quiencle/issues/1 ); 

Should I update the min_zoom and max_zoom of neighbourhoods that I've designated as deprecated and have superceded records? There's about 3 of these neighbourhoods that had values of -1 for min_zoom and 2 additional ones that have min_zoom values of 12. 

Not to slog this out even longer, but there are 3 records (whose deprecated records were within CLE boundaries in WOF) but are outside of Cleveland's boundaries with correct geometry. Let me know how you'd like to proceed with those. Shall I keep them in this file for the time being?

**Comment by nvkelso** [comment]
> min and max zooms for the venues? (There were 2 neighbourhoods in the WOF that I've reclassified as venues, RE: skorasaurus/quiencle#1 );

Those won't show up on the map at this time, don't worry about them.

> Should I update the min_zoom and max_zoom of neighbourhoods that I've designated as deprecated and have superceded records? There's about 3 of these neighbourhoods that had values of -1 for min_zoom and 2 additional ones that have min_zoom values of 12.

No, they will no longer show up on the map (we'll keep the current values as archival only).

> Not to slog this out even longer, but there are 3 records (whose deprecated records were within CLE boundaries in WOF) but are outside of Cleveland's boundaries with their updated geometries. Let me know how you'd like to proceed with those. Shall I keep them in this file for the time being?

Please keep them inside this set for the time being.

**Comment by skorasaurus** [comment]
Thanks, one more question that I want to be sure about before submitting another revision:

Do a place's boundaries need to be completely within the parent's boundary? There are a couple existing WOF records' geometries that go into records which I'm not touching because I don't know enough about that particular neighbourhood and defer to existing records. 

There's a couple other records that had gone outside of Cleveland's boundaries that I had modified (where the neighborhood boundary stops at the boundary, which generally is not always the case..).

**Comment by stepps00** [comment]
Hey @skorasaurus - A record's boundaries don't _have_ to fall completely within the parent's boundary, no. Generally, though, we try to keep them within a parent's boundary.

If you do decide that a record should fall under two parents, it's `parent_id` should be given a value of `-3` to represent multiple parents. A short description of `parent_id` issues can be found at the bottom of [this page](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/how_to/requirements_for_s3.md).

**Comment by skorasaurus** [comment]
Hi, 

Ready for another review at https://github.com/skorasaurus/quiencle/commit/3aee6f45ee3b00f89c91ca14ef93511c9e8d6ca2 

Just to be on the safe side, I've created a branch in my repo - [ohsnap](https://github.com/skorasaurus/quiencle/commit/3761c5db030e0202690b0fd6039a7beca9e4782c). There, I'll be snapping the geometries and then after the snapping is complete, make geojson files based on the [comments made by Stephen in an earlier comment in the issue](https://github.com/whosonfirst-data/whosonfirst-data/issues/189#issuecomment-244193556).

**Comment by stepps00** [comment]
Hey @skorasaurus, thanks for the new data!

I took a look at the new geojson and a have a couple comments:

* I know you're still working on the snapping, but just to note: the screenshot below shows the neighbourhood of Barber Vega Queen (and surrounding areas). Notice the sharp angles and gaps between shapes:

<img width="1029" alt="screen shot 2016-09-15 at 4 02 15 pm" src="https://cloud.githubusercontent.com/assets/18567700/18570456/e2a84752-7b5e-11e6-93af-9d8adfd3c6f5.png">

Probably good practice to go through each shape and make sure all vertices are snapped before finalizing your geojsons by `wof:placetype`

* Neighbourhood vs. neighborhood - a minor edit needed to two "neighborhood" placetype values

* It looks like a few of the features in your geojson fall outside of the Cleveland city boundary. Do you have a plan for these features?

Thanks for the progress so far! What are your planned next steps?

**Comment by skorasaurus** [comment]
Hey, sorry got busy but I finished snapping what I believe to be all of them; it's in my latest commit. 

As far as what to do with the ones to do outside the city; I've left them in here, deferring to @nvkelso 's comment at https://github.com/whosonfirst-data/whosonfirst-data/issues/189#issuecomment-245381583

**Comment by stepps00** [comment]
Hey @skorasaurus - thanks for updating these features. All in all, the geojson is looking great! A few comments/questions:

* `placetype` values: It looks like "neighborhood" is being used instead of "neighbourhood" for a couple of features. This is a minor edit needed if these shapes are going into a PR.

* Features outside of Cleveland can be removed from the geojson (or PR) if they are not being edited. If they are being edited (deprecated, changing parent_id, etc), then go ahead and leave them in.

* In a few cases - Goodrich Kirtland and Chinatown, for example - it looks like a neighbourhood falls completely within another neighbourhood. Should one be a microhood parented by the other? Should each of these be separate neighbourhoods without any overlap? Same goes for East Bank and The Flats.

* In a few cases, like Cudell, it looks like WOF has a current record for the feature, but a new feature was created. The original record was deprecated and superseding work was done between the two. We should normally keep the existing record and just update the geometries - how did you decide on deprecating features in cases like Cudell? 

* Once you think the geojson file is complete and ready for import, let me know - wewill have to figure out an import strategy (likely a PR and separate geojson files for each placetype and case).

All in all, great stuff - I'm excited to see what happens in Cleveland! Thanks for the progress update.

**Comment by skorasaurus** [comment]
>  placetype values: It looks like "neighborhood" is being used instead of "neighbourhood" for a couple of features. This is a minor edit needed if these shapes are going into a PR.

A mistake on my part, fixed in https://github.com/skorasaurus/quiencle/commit/293b5c0e18702ef87b0d5638983d8ba5144e3aba

>   Features outside of Cleveland can be removed from the geojson (or PR) if they are not being edited. If they are being edited (deprecated, changing parent_id, etc), then go ahead and leave them in.

They were edited. In 3 cases (Nela Park, Highland House, Edgewater Towers), they had incorrect geometry and were placed within the city of the Cleveland. In the remaining 4th case, it's a neighbourhood (Gold Coast) that is bordering Cleveland and is the parent of two of geometries (Highland House, Edgewater Towers) that were included. 

>     In a few cases - Goodrich Kirtland and Chinatown, for example - it looks like a neighbourhood falls completely within another neighbourhood. Should one be a microhood parented by the other? Should each of these be separate neighbourhoods without any overlap? Same goes for East Bank and The Flats. 

Yes, Asiatown (Chinatown is the old, deprecated name), is a microhood of GK. I've fixed this in https://github.com/skorasaurus/quiencle/commit/9e87ff896f33ad538739ce45a4aa43c344fdf59f 

Regarding the Flats and East Bank; the East Bank is a part of The Flats and is a special case in general (an S3 case), I can explain more when you'd like. 

>     In a few cases, like Cudell, it looks like WOF has a current record for the feature, but a new feature was created. The original record was deprecated and superseding work was done between the two. We should normally keep the existing record and just update the geometries - how did you decide on deprecating features in cases like Cudell?

In Cleveland, I made significant events for places that had a change in place_type, a parent_id, or name. 
Regarding Cudell and a few others: as I interpreted the then-definition of the [Significant Events](https://github.com/whosonfirst/whosonfirst-cookbook/blob/a444208df17da1239696e85881b404efdb94b6d7/definition/significant_event.md), 
I erred on the side of caution and attributed records whose geometries were modified under 50% as significant events and made new records for them. I do see that this 'gray area' has now been removed from the definition.

**Comment by stepps00** [comment]
Hey @skorasaurus! 

Thanks for putting in the work to update these features and respond to the comments. I will take a look at these shapes and let you know if there are any last changes that need to be made - I may do this early next week, but keep an eye on this issue.

And yes - I've made some changes to our documentation (specifically the significant events and life cycle documents), so the "grey area" is no longer relevant. 

Thanks again - I'll update you when I take a look at the geojson files!

**Comment by nvkelso** [comment]
@skorasaurus This is looking so awesome! Thanks for putting the Cleveland neighbourhoods together :)

I did notice one area downtown that is slightly confusing (image below).

@stepps00 we faced similar issues in Seattle and SF downtowns. Do we need a filler Downtown neighbourhood features as well? Recommendations?

<img width="642" alt="screen shot 2016-10-05 at 13 14 45" src="https://cloud.githubusercontent.com/assets/853051/19129854/b78247a4-8afd-11e6-8717-b643be2d960b.png">

**Comment by stepps00** [comment]
@nvkelso - we do, we shouldn't have gaps in neighbourhood coverage. I'd suggest filling in this gap with a "Downtown" neighbourhood feature.. I'll take a look at the shapes shortly.

cc @skorasaurus

**Comment by skorasaurus** [comment]
I was aware of this as I was editing the data... There is a macrohood for the area highlighted in red.  I was a little befuddled what to put there, because the specific areas not marked in blue are usually just referred to (and known as) as Downtown (with other distinct parts of downtown named as the Microhoods) and have various land uses (the airport, coast guard, some old warehouses/industry, general office...). It's referred to as Downtown by the City Planning for it's worth as well.

**Comment by nvkelso** [comment]
Should Downtown be neighbourhood alone, with microhoods. Which features
there require further nesting?

**Comment by skorasaurus** [comment]
My question to you is: Are places like 'Downtown' a neighbourhood only? I had imagined them to be macrohoods in some contexts including in Cleveland thus, designated Downtown as a macro in this case. 
If you think it should be a

**Comment by stepps00** [comment]
In this case, Downtown should probably be a neighbourhood record as well as a macrohood record.
The macrohood record would parent the Downtown neighbourhood record in addition to Civic Center, the Warehouse District, etc.

We are importing a similar case in San Francisco - it is OK to have a records that share the same name but have different placetypes. The bigger issue is making sure we don't have any geometry gaps.

Thanks again, things are looking good in Cleveland!

**Comment by skorasaurus** [comment]
Thanks for clarification; so create a new geometry with neighbourhood place_type and have it consist of the area not shaded in blue in kelso's post above?

**Comment by stepps00** [comment]
Yeah, that is fine - as long as we're confident that the area not shaded in blue is considered "downtown" (only pointing this out because I am not entirely familiar with Cleveland).

**Comment by skorasaurus** [comment]
Added that downtown neighbourhood (and yes, I'm confident the area not shaded in blue is downtown). 

With my latest commit - https://github.com/skorasaurus/quiencle/commit/de9e70f8dd8c3ec205b014a1f6811a5677c79e21 - my
only lingering question before splitting them into geojson files for import is the following: 

Columns like is_landuse_aoi , is_funky, is_hard_boundary, had 0, -1, or null as its value; 
For features that don't have these columns apply to them ([as Stephen described above, I don't believe there's any of these cases in Cleveland](https://github.com/whosonfirst-data/whosonfirst-data/issues/189#issuecomment-244193556)), should we assign a consistent value to all of the features that are not deprecated?

**Comment by skorasaurus** [comment]
I went back to this today and figured out how to split the geojson files by attributes. My remaining question is how do I generate the lbl_longitude and lbl_latitude? I read the [cookbook](https://github.com/whosonfirst/whosonfirst-cookbook/blob/ae31dc0fcd5c19c45ea795839d4394cbe90e6943/issue_workflows/sf_neighbourhood_updates_pt_1.md) but it states just to run it through mapshaper which I did (without any options) on caution, but the lbl properties' values weren't included. I'm thinking that I need to some additional options, but not sure which ones are you specifying (see https://github.com/mbloch/mapshaper/wiki/Command-Reference for reference).

**Comment by stepps00** [comment]
Hmm.. the `lbl:latitude` and 'lbl:longitude` are derived from MapShaper.. the command line tool can be a bit tricky, but there is also aMapShaper website that allows you to upload files to generate lat/longs... [http://mapshaper.org/](http://mapshaper.org/). Let me know if you're still having difficulties.

**Comment by skorasaurus** [comment]
I figured out how to add the lbl_latitude and lbl_longitude values to each geometry inside the geojson file with mapshaper. Since it's not very intuitive, I added a draft of instructions to a [Pull Request for the cookbook](https://github.com/whosonfirst/whosonfirst-cookbook/pull/6) 

I noticed the original file that you gave me has a property lbl_latitude ; not lbl:latitude ; this was the case for lbl:longitude as well so I wrote the pull request with that in mind. 

It just now needs the is_current and cessation fields attached to it, which if you'd like me to do, let me know or if you'll cover it as [you offered to earlier](https://github.com/whosonfirst-data/whosonfirst-data/issues/189#issuecomment-239864911), that'd be appreciated. 

I've added the files in https://github.com/skorasaurus/quiencle/commit/0a402301ad8e3d5d4254856fabc0d92943eb89d0 and would appreciate a final review.

**Comment by stepps00** [comment]
Great! Thanks for the PR - I'll take a look at that in just a moment... and sorry for any confusion with the `lbl_latitude` & `lbl_longitude` fields.

I'll check out your final geojson files this afternoon and reply here with any questions or comments I have. Thanks again for all the work so far!

**Comment by stepps00** [comment]
Hey @skorasaurus - haven't forgotten about the Cleveland neighbourhood updates.. we've started importing new and updated neighbourhood features into Who's On First (see: [Oakland]() or [San Diego]() for examples) and now have a process to import new features.

I'm going to format your GeoJSON files locally, review, update, and merge into `master` sometime in the near future... stay tuned. I'll update this issue if I have any questions/comments.

**Comment by skorasaurus** [comment]
Hi @stepps00 - I was just following up to see if there's anything I could do to help this move along; I'm considering using mapzen tiles on a Cleveland-centric project and would appreciate it to see these changes incorporated into the vector data. 

(I totally understand y'all are very busy, so no worries. :))

**Comment by stepps00** [comment]
hey @skorasaurus - thanks for following up! I've added this issue to a priority list of tasks (#1002) - I will take a look at the updates early next week and import the changes.

**Comment by stepps00** [comment]
@skorasaurus - I cloned your https://github.com/skorasaurus/quiencle repo and poked around the data, a few last questions.

It looks like the neighbourhood files are "New_Neighbourhoods.geojson" (green) and "Updated_Neighbourhoods.geojson" (blue). Loading these in QGIS, I see:

<img width="966" alt="screen shot 2017-10-27 at 1 16 50 pm" src="https://user-images.githubusercontent.com/18567700/32123525-6a95f1d2-bb19-11e7-8ab6-e8adb16afc09.png">

The red boundary is what WOF currently maintains as the geometry for [Cleveland](https://whosonfirst.mapzen.com/spelunker/id/101712563/). You'll notice gaps in coverage and several polygons that fall outside of the boundary of Cleveland - do you have any data to fill in these gaps?

If not, I can reconcile your edits with an open source dataset of Cleveland neighbourhoods. There are a few minor snapping and property adjustments, but I can easily clean those up if we are able to fill in the "gaps". Thanks!
